core/tests/test_views.py

```diff
@@ -37,27 +40,46 @@ def community() -> Community:
 
 
 def test_add_post_valid(client: Client, user: User, community: Community) -> None:
-    data = {
-        "community": community.pk,
-        "title": "Test Post Title",
-        "content": "This is a test post content.",
-    }
-    client.force_login(user)
-    response = client.post(reverse("post-create"), data=data, follow=True)
-    assert response.status_code == 200
-    assert response.context["post"].author == user
-    assert response.context["post"].title == data["title"]
-    assert response.context["post"].content == data["content"]
+    with tempfile.NamedTemporaryFile(suffix=".jpg", delete=False) as tmp_file:
+        image = Image.new("RGB", (100, 100), color=(73, 109, 137))
+        image.save(tmp_file, format="JPEG")
+        tmp_file.seek(0)
+
+        avatar_file = SimpleUploadedFile(name="test_avatar.jpg", content=tmp_file.read(), content_type="image/jpeg")
+
+        user.avatar = avatar_file
+        user.save()
+
+        data = {
+            "community": community.pk,
+            "title": "Test Post Title",
+            "content": "This is a test post content.",
+        }
+        client.force_login(user)
+        response = client.post(reverse("post-create"), data=data, follow=True)
+        assert response.status_code == 200
+        assert response.context["post"].author == user
+        assert response.context["post"].title == data["title"]
+        assert response.context["post"].content == data["content"]
 
 
 def test_add_post_invalid(client: Client, user: User, community: Community) -> None:
-    data = {"title": ""}
-    client.force_login(user)
-    response = client.post(reverse("post-create"), data=data)
-    assert response.status_code == 200
-    form = response.context["form"]
-    assert "This field is required." in form.errors["title"]
-    assert "This field is required." in form.errors["content"]
+    with tempfile.NamedTemporaryFile(suffix=".jpg", delete=False) as tmp_file:
```

Comment:
This is a duplicate of code from line 43. DRY (don't repeat yourself). Please extract this to a function and use it as a function to add avatar. Or simply drop this image here. Testing user avatar when post is not added does not make sense, the avatar is not displayed on this post, no?

Reply:
Done, Okey